examples/official-site/your-first-sql-website/extensions-to-sql.md

```diff
@@ -0,0 +1,209 @@
+# Extensions to SQL
+
+SQLPage makes some special treatment before executing your SQL queries.
+
+When executing your SQL file, SQLPage executes each query one at a time.
+It doesn't send the whole file as-is to the database engine.
+
+## Performance
+
+See the [performance page](/performance.sql) for details on the optimizations
+made to run your queries as fast as possible.
+
+## Variables
+
+SQL doesn't have its own mechanism for variables.
+SQLPage implements variables in the following way:
+
+### POST parameters
+
+When sending a POST request, most often by sending a form with the
+[form component](/component.sql?component=form), the form data is made
+available as variables prefixed by a semi-colon.
+
+So when this form is sent:
+
+`form.sql`
+```sql
+SELECT
+    'form' AS component,
+    'POST' AS method, -- form defaults to using the HTTP POST method
+    'result.sql' AS action;
+
+SELECT
+    'age' AS name,
+    'How old are you?' AS label,
+    'number' AS type;
+```
+
+It will make a request to this page:
+
+`result.sql`
+```sql
+SELECT
+    'text' AS component,
+    'You are ' || :age || ' years old!' AS contents;
+```
+
+`:age` will be substituted by the actual value of the POST parameter.
+
+### URL parameters
+
+Likewise, URL parameters are available as variables prefixed by a dollar sign.
+
+> URL parameters are often called GET parameters because they can originate
+> from a form with 'GET' as the method.
+
+So the previous example can be reworked to handle URL parameters:
+
+`result.sql`
+```sql
+SELECT
+    'text' AS component,
+    'You are ' || $age || ' years old!' AS contents;
+```
+
+By querying this page with this URL: `/request.sql?age=42`
+we would get `You are 42 years old!` as a response.
+
+### The `SET` command
+
+SQLPage adds a non-standard SQL keyword: `SET`.
```

Comment:
well `SET` already exists in most sql implementations. Maybe we could formulate this as : 

```suggestion
SQLPage overrides the behavior of `SET` statements in SQL to store variables in SQLPage itself instead of running the statement on the database. 
```